Model/PayolutionInvoicePaymentMethod.php

```diff
@@ -171,10 +287,59 @@ public function getPublicFieldNames()
      */
     public function getMetaDataFieldNames()
     {
-        return [
+        $aReturn = [
+            'allowed_currencies',
             'shipping_countries',
             'billing_countries',
             'billing_shipping',
         ];
+
+        $aCurrencies = PaymentMethodHelper::getCurrencyOptions();
+        $aFieldNames = ['httpuser', 'httppass', 'maid', 'secret'];
+
+        foreach ($aCurrencies as $sCurrency) {
+            foreach ($aFieldNames as $sFieldName) {
+                $aReturn[] = $sFieldName . '_' . strtolower($sCurrency);
+            }
+        }
+
+        return $aReturn;
+    }
+
+    /**
+     * @inheritdoc
+     *
+     * @param PayolutionInvoiceTransaction $oTransaction
+     * @param Order                        $oOrder
+     *
+     * @since 1.2.0
+     */
+    public function addMandatoryTransactionData(&$oTransaction, $oOrder)
+    {
+        parent::addMandatoryTransactionData($oTransaction, $oOrder);
+
+        $oAccountHolder = $oOrder->getAccountHolder();
+        $oAccountHolder->setDateOfBirth(new DateTime(SessionHelper::getDbDateOfBirth(self::getName())));
+
+        if (SessionHelper::isPhoneValid(self::getName())) {
+            $oAccountHolder->setPhone(SessionHelper::getPhone(self::getName()));
+        }
+
+        $oTransaction->setAccountHolder($oAccountHolder);
+    }
+
+    /**
+     * @inheritdoc
+     *
+     * @return bool
+     *
+     * @since 1.2.0
+     */
+    protected function _isPhoneMandatory()
+    {
+        $oBillingCountry = oxNew(Country::class);
+        $oBillingCountry->load(Registry::getSession()->getUser()->oxuser__oxcountryid->value);
+
+        return $oBillingCountry->oxcountry__oxisoalpha2->value === 'NL';
```

Comment:
`NL` should probably be in a const. And I'd make it an array so that it could theoretically contain multiple country codes.